### ISSUE TITLE
How to handle speed up (dropped/replaced) transactions?

### ISSUE THREAD

**Issue by quagliero** [EthWorks/useDApp#307]
I'm wondering if there is anything in-built or any guidance on handling when a user speeds up a transaction? The `state` object of the useContractFunction no longer matches and it seems to just stay in `Mining`.

**Comment by quagliero** [comment]
Just to add to this, we've implemented an update on the NFTX site to handle sped-up transactions by having our own version of the `useContractFunction` that uses an updated version of `usePromiseTransaction`. The only change was adding an update into the `catch` block of the useCallback (https://github.com/EthWorks/useDApp/blob/master/packages/core/src/hooks/usePromiseTransaction.ts#L29-L37).

In case anyone else wants to do the same, the updated `catch` block is:

```ts
      } catch (e) {
        const errorMessage =
          e.error?.message ?? e.reason ?? e.data?.message ?? e.message;

        if (transaction) {
          let status: TransactionState = 'Fail';
          // transaction has been dropped&replaced but *NOT* cancelled (user has sped up the tx)
          if (
            e?.code === 'TRANSACTION_REPLACED' &&
            e?.replacement &&
            (e?.reason === 'repriced' || e?.cancelled === false)
          ) {
            const type =
              e.receipt.status === 0
                ? 'transactionFailed'
                : 'transactionSucceed';

            // was replace transaction a success or fail
            status = type === 'transactionSucceed' ? 'Success' : 'Fail';

            // notify of this new tx
            addNotification({
              notification: {
                type,
                submittedAt: Date.now(),
                transaction: e?.replacement,
                receipt: e.receipt,
                transactionName: e?.replacement?.transactionName,
              },
              chainId,
            });
          }

          setState({
            status,
            transaction: e?.replacement || transaction,
            receipt: e.receipt,
            errorMessage,
            chainId,
          });
        } else {
          setState({ status: 'Exception', errorMessage, chainId });
        }
        return undefined;
      }
```

**Comment by tt-marek** [comment]
Support for dropped/replaced transactions merged in #355
